rocketpy/tools.py

```diff
@@ -23,3 +26,978 @@ def __get__(self, instance, owner=None):
+
+        Notes
+        -----
+        The cross product between two vectors can be computed as the matrix
+        product between the cross matrix of the first vector and the second
+        vector.
+
+        Examples
+        --------
+        >>> v = Vector([1, 7, 3])
+        >>> u = Vector([2, 5, 6])
+        >>> (v ^ u) == v.cross_matrix @ u
+        True
+        """
+        return Matrix(
+            [[0, -self.z, self.y], [self.z, 0, -self.x], [-self.y, self.x, 0]]
+        )
+
+    def __abs__(self):
+        """R3 vector norm, magnitude or absolute value."""
+        return (self.x**2 + self.y**2 + self.z**2) ** 0.5
+
+    def __neg__(self):
+        """-1 times R3 vector self."""
+        return Vector([-self.x, -self.y, -self.z])
+
+    def __add__(self, other):
+        """Sum two R3 vectors."""
+        result = [self[i] + other[i] for i in range(3)]
+        return Vector(result)
+
+    def __radd__(self, other):
+        """Sum two R3 vectors."""
+        return self.__add__(other)
+
+    def __sub__(self, other):
+        """Subtract two R3 vectors."""
+        result = [self[i] - other[i] for i in range(3)]
+        return Vector(result)
+
+    def __rsub__(self, other):
+        """Subtract two R3 vectors."""
+        result = [other[i] - self[i] for i in range(3)]
+        return Vector(result)
+
+    def __mul__(self, other):
+        """Component wise multiplication between R3 vector and scalar other."""
+        return self.__rmul__(other)
+
+    def __rmul__(self, other):
+        """Component wise multiplication between R3 vector and scalar other."""
+        return Vector([other * self.x, other * self.y, other * self.z])
+
+    def __truediv__(self, other):
+        """Component wise division between R3 vector and scalar other."""
+        return Vector([self.x / other, self.y / other, self.z / other])
+
+    def __xor__(self, other):
+        """Cross product between self and other.
+
+        Parameters
+        ----------
+        other : Vector
+            R3 vector to be crossed with self.
+
+        Returns
+        -------
+        Vector
+            R3 vector resulting from the cross product between self and other.
+
+        Examples
+        --------
+        >>> v = Vector([1, 7, 3])
+        >>> u = Vector([2, 5, 6])
+        >>> (v ^ u)
+        Vector(27, 0, -9)
+
+        Notes
+        -----
+        Parameters order matters, since cross product is not commutative.
+        Parentheses are required when using cross product with the ^ operator
+        to avoid ambiguity with the bitwise xor operator and keep the
+        precedence of the operators.
+        """
+        x = self[1] * other[2] - self[2] * other[1]
+        y = -self[0] * other[2] + self[2] * other[0]
+        z = self[0] * other[1] - self[1] * other[0]
+        return Vector([x, y, z])
+
+    def __matmul__(self, other):
+        """Dot product between two R3 vectors."""
+        return sum([self[i] * other[i] for i in range(3)])
```

Comment:
There is also a roughly 2x speed increase in performing:

```py
return self[0]*other[0] + self[1]*other[1] + self[2]*other[2]
```

The same comment as above about the usage of `self.x` applies.